### PR TITLE
ctrl add config & test p0 & fix modeling bug

### PR DESCRIPTION
**Pull request by wjx-sudo**
### PR types

ctrl add config & test p0 & fix modeling bug
### PR changes

Models

### CODE REVIEW

File: paddlenlp/transformers/ctrl/modeling.py [hunk 1]
```diff
@@ -708,12 +675,12 @@ class CTRLForSequenceClassification(CTRLPreTrainedModel):
...
-        super().__init__()
-        self.num_classes = num_classes
-        self.ctrl = ctrl
-        self.dropout = nn.Dropout(dropout if dropout is not None else self.ctrl.config["hidden_dropout_prob"])
-        self.classifier = nn.Linear(self.ctrl.config["hidden_size"], num_classes, bias_attr=False)
+    def __init__(self, config: CTRLConfig):
+        super().__init__(config)
+        self.num_classes = config.num_classes
+        self.ctrl = CTRLModel(config)
+        # self.dropout = nn.Dropout(config.dropout if config.dropout is not None else config.hidden_dropout_prob)
```
Comment (sijunhe) [inline paddlenlp/transformers/ctrl/modeling.py]: 这个怎么关了

File: paddlenlp/transformers/ctrl/configuration.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    >>> configuration = CTRLConfig()
+    >>> # Initializing a model (with random weights) from the configuration
+    >>> model = CTRLModel(configuration)
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    pretrained_init_configuration = CTRL_PRETRAINED_INIT_CONFIGURATION
+    model_type = "ctrl"
+    keys_to_ignore_at_inference = ["past_key_values"]
```
Comment (sijunhe) [inline paddlenlp/transformers/ctrl/configuration.py]: 我们没有这个功能，不需要和HF对齐

File: tests/transformers/ctrl/test_modeling.py [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+            n_embd=self.hidden_size,
+            n_layer=self.num_hidden_layers,
+            n_head=self.num_attention_heads,
+            # intermediate_size=self.intermediate_size,
+            # hidden_act=self.hidden_act,
+            # hidden_dropout_prob=self.hidden_dropout_prob,
+            # attention_probs_dropout_prob=self.attention_probs_dropout_prob,
+            n_positions=self.max_position_embeddings,
+            # type_vocab_size=self.type_vocab_size,
+            # initializer_range=self.initializer_range,
```
Comment (sijunhe) [inline tests/transformers/ctrl/test_modeling.py]: 为啥注释掉？

File: paddlenlp/transformers/ctrl/modeling.py (hunk 1 above)
Comment (wjx-sudo) [reply paddlenlp/transformers/ctrl/modeling.py]: dropout 在该模型forword里边没有用到；而且config里边没有dropout 和 hidden_dropout_prob;预训练模型config里面也没看见有这个属性；
考虑是不是可以删掉。

File: paddlenlp/transformers/ctrl/modeling.py [hunk 4]
```diff
@@ -540,23 +472,23 @@ class CTRLLMHeadModel(CTRLPreTrainedModel):
...
 
     """
 
-    def __init__(self, ctrl):
-        super().__init__()
-        self.ctrl = ctrl
+    def __init__(self, config: CTRLConfig):
+        super().__init__(config)
+        self.ctrl = CTRLModel(config)
         if self.ctrl.config["tie_word_embeddings"]:
```
Comment (sijunhe) [inline paddlenlp/transformers/ctrl/modeling.py]:
```suggestion
        if config.tie_word_embeddings:
```

File: tests/transformers/ctrl/test_modeling.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+        model = CTRLForSequenceClassification(config)
+        model.eval()
+        sequence_labels = ids_tensor([self.batch_size], self.type_sequence_label_size)
+        result = model(input_ids=input_ids, token_type_ids=token_type_ids, labels=sequence_labels)
+        self.parent.assertEqual(len(result), 2)
+        self.parent.assertEqual(result[1].shape, [self.batch_size, self.num_labels])
+
+
+@parameterized_class(
+    ("return_dict", "use_labels"),
```
Comment (sijunhe) [inline tests/transformers/ctrl/test_modeling.py]: 没有return_dict 可以删掉

File: paddlenlp/transformers/ctrl/modeling.py (hunk 1 above)
Comment (sijunhe) [reply paddlenlp/transformers/ctrl/modeling.py]: ok

**Review by sijunhe — COMMENTED** [review]
各种被注释掉的代码清理一下，主要在test_modeling里面

**Review by sijunhe — APPROVED** [review]
lgtm
